test/test_entity.rb

```diff
@@ -104,6 +102,62 @@ def test_replace_entities
       assert_equal source, out
     end
 
+    def test_entity_with_reference
+      entity = REXML::Entity.new( [:entitydecl, "a", "B", "%"])
+      assert_equal('<!ENTITY % a "B">', entity.to_s)
+      assert_equal("a", entity.name)
+      assert_equal("B", entity.value)
+      assert_equal("B", entity.normalized)
+      assert_equal("B", entity.unnormalized)
```

Comment:
Could you check by one `assert_equal` for easy to debug?
FYI: https://www.clear-code.com/blog/2011/2/28.html

```suggestion
      assert_equal([
                     '<!ENTITY % a "B">',
                     "a",
                     "B",
                     "B",
                     "B",
                   ],
                   [
                     entity.to_s,
                     entity.name,
                     entity.value,
                     entity.normalized,
                     entity.unnormalized,
                   ])
```